$NetBSD$

The task-controller and jsvc components only build on Linux. The librecordio
component doesn't like to build against xerces-c (recent GCC too aggressive?).
Simplify installation phase for pkgsrc purposes.
--- build.xml.orig	2012-10-03 05:17:24.000000000 +0000
+++ build.xml
@@ -1595,14 +1595,6 @@
 
     <copy todir="${dist.dir}/" file="build.xml"/>
 
-    <subant target="task-controller">
-      <fileset dir="." includes="build.xml"/>
-    </subant>
-
-    <subant target="jsvc">
-      <fileset dir="." includes="build.xml"/>
-    </subant>
-
     <chmod perm="ugo+x" type="file" parallel="false">
         <fileset dir="${dist.dir}/bin"/>
         <fileset dir="${dist.dir}/sbin"/>
@@ -1667,17 +1659,16 @@
     </macro_tar>
   </target>
 
-  <target name="bin-package" depends="compile, jar, examples, tools-jar, jar-test, ant-tasks, package-librecordio" 
+  <target name="bin-package" depends="compile, jar, examples, tools-jar, jar-test, ant-tasks"
 		description="assembles platform-specific artifacts for binary targets">
     <delete dir="${dist.dir}"/>
     <mkdir dir="${dist.dir}"/>
     <mkdir dir="${dist.dir}/bin"/>
-    <mkdir dir="${dist.dir}/etc/hadoop"/>
     <mkdir dir="${dist.dir}/lib"/>
-    <mkdir dir="${dist.dir}/lib64"/>
     <mkdir dir="${dist.dir}/libexec"/>
     <mkdir dir="${dist.dir}/sbin"/>
     <mkdir dir="${dist.dir}/share/${name}/contrib"/>
+    <mkdir dir="${dist.dir}/share/${name}/examples"/>
     <mkdir dir="${dist.dir}/share/${name}/webapps"/>
     <mkdir dir="${dist.dir}/share/${name}/templates/conf"/>
 
@@ -1706,7 +1697,7 @@
     <exec dir="${dist.dir}" executable="sh" failonerror="true">
       <env key="BASE_NATIVE_LIB_DIR" value="${lib.dir}/native"/>
       <env key="BUILD_NATIVE_DIR" value="${build.dir}/native"/>
-      <env key="DIST_LIB_DIR" value="${dist.dir}/${lib.data.model}"/>
+      <env key="DIST_LIB_DIR" value="${dist.dir}/lib"/>
       <env key="NATIVE_PLATFORM" value="${build.platform}" />
       <arg line="${native.src.dir}/packageBinNativeHadoop.sh"/>
     </exec>
@@ -1743,7 +1734,7 @@
       </fileset>
     </copy>
 
-    <copy todir="${dist.dir}/etc/hadoop">
+    <copy todir="${dist.dir}/share/examples/hadoop">
       <fileset dir="${conf.dir}" excludes="**/*.template"/>
     </copy>
 
@@ -1765,19 +1756,16 @@
       </fileset>
     </copy>
 
-    <copy todir="${dist.dir}/${lib.data.model}" includeEmptyDirs="false" flatten="true">
+    <copy todir="${dist.dir}/lib" includeEmptyDirs="false" flatten="true">
       <fileset dir="${build.dir}/c++">
-        <include name="${build.platform}/lib/*"/>
+        <include name="${build.platform}/lib/libhadoop*.a"/>
+        <include name="${build.platform}/lib/libhdfs.la"/>
+        <include name="${build.platform}/lib/libhdfs.so.0.0.0"/>
       </fileset>
     </copy>
 
-    <subant target="task-controller">
-      <fileset dir="." includes="build.xml"/>
-    </subant>
-
-    <subant target="jsvc">
-      <fileset dir="." includes="build.xml"/>
-    </subant>
+    <symlink link="${dist.dir}/lib/libhdfs.so.0" resource="libhdfs.so.0.0.0"/>
+    <symlink link="${dist.dir}/lib/libhdfs.so" resource="libhdfs.so.0.0.0"/>
 
     <chmod perm="ugo+x" type="file" parallel="false">
         <fileset dir="${dist.dir}/bin"/>
